src/components/notification/inline-notification.ts

```diff
@@ -83,24 +83,49 @@ const iconsForKinds = {
  */
 @customElement(`${prefix}-inline-notification`)
 class BXInlineNotification extends FocusMixin(LitElement) {
+  /**
+   * Current timeout identifier
+   */
+  protected _timeoutID: number | null = null;
+
   /**
    * Notification type.
    */
   protected _type = NOTIFICATION_TYPE.INLINE;
 
+  /**
+   * Cancels the current timeout configured for the notification
+   * @param timeoutID current timeout's identifier
+   */
+  protected _cancelTimeout(timeoutID: number) {
+    clearTimeout(timeoutID);
+    this._timeoutID = null;
+  }
+
+  /**
+   * Overrides (if exists) the timeout to close the notification
+   * @param timeout the time in ms
+   */
+  protected _initializeTimeout(timeout: number) {
+    if (this._timeoutID) {
+      this._cancelTimeout(this._timeoutID);
+    }
+    this._timeoutID = setTimeout(this._handleUserOrTimerInitiatedClose.bind(this), timeout);
+  }
+
   /**
    * Handles `click` event on the close button.
    * @param event The event.
    */
   protected _handleClickCloseButton({ target }: MouseEvent) {
-    this._handleUserInitiatedClose(target);
+    this._handleUserOrTimerInitiatedClose(target);
   }
 
   /**
-   * Handles user-initiated close request of this modal.
-   * @param triggeredBy The element that triggered this close request.
+   * Handles user-initiated or through timer close request of this modal.
+   * @param triggeredBy The element that triggered this close request, if there is one.
    */
-  protected _handleUserInitiatedClose(triggeredBy: EventTarget | null) {
+  protected _handleUserOrTimerInitiatedClose(triggeredBy?: EventTarget | null) {
```

Comment:
```suggestion
  protected _handleUserOrTimerInitiatedClose(triggeredBy: EventTarget | null) {
```

Reply:
The problem that I see with this is that when the `setTimeout` is launched the value in `triggeredBy` is **undefined** so I think is interesting to specify that the property can be **undefined**. How do you usually handle that?

Reply:
Ok, nevermind. I just saw your last proposal!